Aardvark/ARKEmailBugReporter.m

```diff
@@ -342,169 +336,143 @@ - (NSString *)formattedLogMessagesAttachmentExtension;
 
 #pragma mark - Private Methods
 
-- (void)_stealFirstResponder;
+- (void)_showBugReportPrompt;
 {
-    ARKInvisibleView *invisibleView = [ARKInvisibleView new];
-    invisibleView.layer.opacity = 0.0;
-    [[UIApplication sharedApplication].keyWindow addSubview:invisibleView];
-    [invisibleView becomeFirstResponder];
-    [invisibleView removeFromSuperview];
-}
-
-- (void)_showBugTitleCaptureAlert;
-{
-    /*
-     iOS 8 often fails to transfer the keyboard from a focused text field to a UIAlertView's text field.
-     Transfer first responder to an invisble view when a debug screenshot is captured to make bug filing itself bug-free.
-     */
-    [self _stealFirstResponder];
-    
-    NSString * const title = NSLocalizedString(@"What Went Wrong?", @"Title text for alert asking user to describe a bug they just encountered");
-    NSString * const message = NSLocalizedString(@"Please briefly summarize the issue you just encountered. You’ll be asked for more details later.", @"Subtitle text for alert asking user to describe a bug they just encountered");
-    NSString * const composeReportButtonTitle = NSLocalizedString(@"Compose Report", @"Button title to compose bug report");
-    NSString * const cancelButtonTitle = NSLocalizedString(@"Cancel", @"Button title to not compose a bug report");
-    
-    UIAlertController *const alertController = [UIAlertController alertControllerWithTitle:title message:message preferredStyle:UIAlertControllerStyleAlert];
-    
-    [alertController addAction:[UIAlertAction actionWithTitle:composeReportButtonTitle style:UIAlertActionStyleDefault handler:^(UIAlertAction *action) {
-        UITextField *textfield = [alertController.textFields firstObject];
-        [self _createBugReportWithTitle:textfield.text];
-    }]];
-    
-    [alertController addAction:[UIAlertAction actionWithTitle:cancelButtonTitle style:UIAlertActionStyleDefault handler:NULL]];
-    
-    [alertController addTextFieldWithConfigurationHandler:^(UITextField *textField) {
-        [self _configureAlertTextfield:textField];
+    id <ARKEmailBugReporterPromptingDelegate> const promptPresenter = (self.promptingDelegate ?: [ARKDefaultPromptPresenter new]);
+    [promptPresenter showBugReportingPromptForConfiguration:[self _configurationWithCurrentSettings] completion:^(ARKEmailBugReportConfiguration * _Nullable configuration) {
+        if (configuration != nil) {
+            [self _createBugReportWithConfiguration:configuration];
+        } else {
+            // If the configuration is nil, the callee has signaled that we should not show a bug report. In the future, we can clean up any persisted state here as necessary.
```

Comment:
In the absence of a specific to-do item, I'd omit the `else` and put this comment before the `if`.